File: 73-set-matrix-zeroes/set-matrix-zeroes.java
class Solution {
    public void setZeroes(int[][] matrix) {
        boolean col=true;
        
        for(int i=0;i<matrix.length;i++)
        {
            if(matrix[i][0]==0)
                    {
                        col=false;
                    }
            for(int j=1;j<matrix[0].length;j++)
            {
                if(matrix[i][j]==0)
                {
                   
                    matrix[0][j]=0;
                    matrix[i][0]=0;
                    
                }
            }
        }
         for(int i=matrix.length-1;i>=0;i--)
        {
            for(int j=matrix[0].length-1;j>=1;j--)
            {

                 
                if(matrix[0][j]==0||matrix[i][0]==0)
                {
                     System.out.println(i+","+j);
                    matrix[i][j]=0;
                }
                
            }if(!col)
            {
               
                matrix[i][0]=0;
            }
        }
    }
}

File: 73-set-matrix-zeroes/set-matrix-zeroes.cpp
class Solution {
public:
    void setZeroes(vector<vector<int>>& matrix) {
        if (matrix.empty() || matrix[0].empty()) return;

        
        int m=matrix.size();
        int n=matrix[0].size();
        bool firstRow=false;
        bool firstCol=false;
        int i=0;
        int j=0;
        for(i=0;i<m;i++)
        {
        if(matrix[i][0]==0)
        {
        firstCol=true;
        break;
        }}

        for( i=0;i<n;i++)
        {
        if(matrix[0][i]==0)
        {
        firstRow=true;
        break;
        }}

        for( i=1;i<m;i++)
        {
        for( j=1;j<n;j++)
        {
        if( matrix[i][j]==0)
        {
        matrix[i][0]=0;
        matrix[0][j]=0;
        }
        }
        }
        for( i=1;i<m;i++)
        {
        for( j=1;j<n;j++)
        {
        if( matrix[i][0]==0 ||matrix[0][j]==0)
        {
        matrix[i][j]=0;
        }
        }
        }

        if(firstCol)
        {
        for( i=0;i<m;i++)
        {
        matrix[i][0]=0;
        }
        }
         if(firstRow)
        {
        for( i=0;i<n;i++)
        {
        matrix[0][i]=0;
        }   
        }

    }
};